cmd/main.go

```diff
@@ -2,16 +2,192 @@ package main
 
 import (
 	"context"
+	"flag"
 	"fmt"
+	"net"
 	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
-	"github.com/datawire/ambassador-agent/cmd/agent"
+	"github.com/datawire/ambassador-agent/pkg/agent"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/klog/v2"
+
+	"github.com/datawire/dlib/dgroup"
+	"github.com/datawire/dlib/dlog"
+	"github.com/emissary-ingress/emissary/v3/pkg/busy"
+	"github.com/emissary-ingress/emissary/v3/pkg/logutil"
+)
+
+// internal k8s service
+const (
+	AdminDiagnosticsPort     = 8877
+	DefaultSnapshotURLFmt    = "http://ambassador-admin:%d/snapshot-external"
+	DefaultDiagnosticsURLFmt = "http://ambassador-admin:%d/ambassador/v0/diag/?json=true"
+
+	ExternalSnapshotPort = 8005
+
+	leaseLockName = "ambassador-agent-lease-lock"
 )
 
 func main() {
-	err := agent.Main(context.Background(), "", os.Args[1:]...)
+	ctx := context.Background()
+
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		fmt.Fprintf(os.Stdout, "error executing from argparser: %v", err)
+		dlog.Error(ctx, err.Error())
 		os.Exit(1)
 	}
+	// creates the clientset
+	clientset := kubernetes.NewForConfigOrDie(config)
+
+	agentNamespace := getEnvWithDefault("AGENT_NAMESPACE", "ambassador")
+
+	ambAgent := agent.NewAgent(nil, agent.NewArgoRolloutsGetter, agent.NewSecretsGetter, clientset, agentNamespace)
+
+	// all log things need to happen here because we still allow the agent to run in amb-sidecar
+	// and amb-sidecar should control all the logging if it's kicking off the agent.
+	// this codepath is only hit when the agent is running on its own
+	logLevel := os.Getenv("AES_LOG_LEVEL")
+	// by default, suppress everything except fatal things
+	// the watcher in the agent will spit out a lot of errors because we don't give it rbac to
+	// list secrets initially.
+	klogLevel := 3
+	if logLevel != "" {
+		logrusLevel, err := logutil.ParseLogLevel(logLevel)
+		if err != nil {
+			dlog.Errorf(ctx, "error parsing log level, running with default level: %+v", err)
+		} else {
+			busy.SetLogLevel(logrusLevel)
+		}
+		klogLevel = logutil.LogrusToKLogLevel(logrusLevel)
+	}
+	klogFlags := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	klog.InitFlags(klogFlags)
+	if err := klogFlags.Parse([]string{fmt.Sprintf("-stderrthreshold=%d", klogLevel), "-v=2", "-logtostderr=false"}); err != nil {
+		dlog.Error(ctx, err.Error())
+		os.Exit(1)
+	}
```

Comment:
Do we still need this if we're sticking with dlog?

Reply:
i left it in because it was in the original, but now that i read it, no, we can toss it